### PR TITLE
Add Cryptocom exchange

### PR DESCRIPTION
**Pull request by germanzelentsov**
Pull request contains the implementation of initial Crypto.com support, including some modules such as Candle, Ticker, and GetInstruments specified in https://github.com/bhftbootcamp/CryptoAPIs.jl/issues/5

The following issue will be resolved with this PR:

closes https://github.com/bhftbootcamp/CryptoAPIs.jl/issues/5 Implement initial Crypto.com support

### Pull request checklist

- [x] Did you bump the project version?
- [x] Did you add a description to the Pull Request?
- [ ] Did you add new tests?
- [ ] Did you add reviewers?

### CODE REVIEW

File: examples/Crypto/Spot.jl [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#Crypto/Spot
+#https://exchange-docs.crypto.com/exchange/v1/rest-ws/index.html#introduction
```
Comment (gryumov) [inline examples/Crypto/Spot.jl:2]:
```suggestion
# Crypto/Spot
# https://exchange-docs.crypto.com/exchange/v1/rest-ws/index.html#introduction
```

File: docs/src/pages/Crypto.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Crypto
+
+```@docs
+CryptoAPIs.Crypto.CryptoClient
+CryptoAPIs.Crypto.CryptoAPIError
+CryptoAPIs.Crypto.Data
```
Comment (gryumov) [inline docs/src/pages/Crypto.md:6]:
```suggestion
CryptoAPIs.Cryptocom.CryptoClient
CryptoAPIs.Cryptocom.CryptoAPIError
CryptoAPIs.Cryptocom.Data
```

File: src/Crypto/Spot/API/GetInstruments.jl [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+using CryptoAPIs.Crypto
+using CryptoAPIs.Crypto: Data
+using CryptoAPIs: Maybe, APIsRequest
+
+Base.@kwdef struct GetInstrumentsQuery <: CryptoPublicQuery
+    #__ empty
+end
+
+struct GetInstrumentsStruct <:CryptoData
+    symbol::Maybe{String}
```
Comment (gryumov) [inline src/Crypto/Spot/API/GetInstruments.jl:19]:
```suggestion
    symbol::String
```

File: src/Crypto/Crypto.jl [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    CryptoAccessQuery,
+    CryptoPrivateQuery,
+    CryptoAPIError,
+    CryptoClient,
+    CryptoData
+
+using Serde
+using Dates, NanoDates, Base64, Nettle
+using UUIDs, JSONWebTokens
+
```
Comment (gryumov) [inline src/Crypto/Crypto.jl:14]:
```suggestion

```

File: src/Crypto/Spot/API/Ticker.jl [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+using CryptoAPIs.Crypto
+using CryptoAPIs.Crypto: Data
+using CryptoAPIs: Maybe, APIsRequest
+
+Base.@kwdef struct TickerQuery <: CryptoPublicQuery
+    instrument_name::Maybe{String} = nothing
+end
+
+struct TickerStruct <:CryptoData
+    h::Maybe{String}
```
Comment (gryumov) [inline src/Crypto/Spot/API/Ticker.jl:19]:
```suggestion
    h::Maybe{Float64}
```

File: src/CryptoAPIs.jl [hunk 6]
```diff
@@ -165,5 +165,6 @@ include("Bybit/Bybit.jl")
 include("Coinbase/Coinbase.jl")
 include("Gateio/Gateio.jl")
 include("Upbit/Upbit.jl")
+include("Crypto/Crypto.jl")
```
Comment (gryumov) [inline src/CryptoAPIs.jl:168]:
```suggestion
include("Coinbase/Coinbase.jl")
include("Crypto/Crypto.jl")
include("Gateio/Gateio.jl")
include("Upbit/Upbit.jl")
```

File: src/Cryptocom/Cryptocom.jl [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+- `method::String`: Return method endpoint.
+- `code::String`: Return code.
+- `result::D`: Request result data.
+"""
+struct Data{D<:Maybe{<:AbstractAPIsData}} <: AbstractAPIsData
+    id::Int64
+    method::String
+    code::String
+    result::D
+end
```
Comment (dmitrii-doronin) [inline src/Cryptocom/Cryptocom.jl:37]: Can the API return `null` in result field? If so, can you give an example?

File: src/Cryptocom/Spot/API/Candle.jl [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+using Serde
+using Dates, NanoDates, TimeZones
+
+using CryptoAPIs.Cryptocom
+using CryptoAPIs.Cryptocom: Data
+using CryptoAPIs: Maybe, APIsRequest
+
+Base.@kwdef struct CandleQuery <: CryptocomPublicQuery
+    instrument_name::String
+    timeframe::Maybe{String} = nothing
```
Comment (dmitrii-doronin) [inline src/Cryptocom/Spot/API/Candle.jl:16]: Why have you decided to skip enums here?

File: src/Cryptocom/Errors.jl [hunk 9]
```diff
@@ -0,0 +1,44 @@
+# Cryptocom/Errors
+# https://exchange-docs.crypto.com/exchange/v1/rest-ws/index.html#response-and-reason-codes
+
+import ..CryptoAPIs: APIsResult, APIsUndefError, isretriable, retry_maxcount, retry_timeout
+
+# UNHANDLED
```
Comment (dmitrii-doronin) [inline src/Cryptocom/Errors.jl:6]: `UNDEF` or `UNHANDLED`?

File: src/Cryptocom/Spot/API/Candle.jl [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    end_ts::Maybe{Int64} = nothing
+end
+
+struct CandleStruct <: CryptocomData
+  o::Maybe{Float64}
+  h::Maybe{Float64}
+  l::Maybe{Float64}
+  c::Maybe{Float64}
+  v::Maybe{Float64}
+  t::Maybe{Int64}
```
Comment (dmitrii-doronin) [inline src/Cryptocom/Spot/API/Candle.jl:28]: It's a unix timestamp. You can do direct conversion to NanoDate here.

Comment (dmitrii-doronin) [inline src/Cryptocom/Spot/API/Candle.jl:28]: It's very dangerous to do that. Is it really a plausible response?

File: src/Cryptocom/Spot/API/Candle.jl [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+using CryptoAPIs.Cryptocom
+using CryptoAPIs.Cryptocom: Data
+using CryptoAPIs: Maybe, APIsRequest
+
+Base.@kwdef struct CandleQuery <: CryptocomPublicQuery
+    instrument_name::String
+    timeframe::Maybe{String} = nothing
+    count::Maybe{Int64} = nothing
+    start_ts::Maybe{Int64} = nothing
+    end_ts::Maybe{Int64} = nothing
```
Comment (dmitrii-doronin) [inline src/Cryptocom/Spot/API/Candle.jl:19]: Should be NanoDate here

File: src/Cryptocom/Spot/API/GetInstruments.jl [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+using CryptoAPIs.Cryptocom
+using CryptoAPIs.Cryptocom: Data
+using CryptoAPIs: Maybe, APIsRequest
+
+Base.@kwdef struct GetInstrumentsQuery <: CryptocomPublicQuery
+    #__ empty
+end
+
+struct GetInstrumentsStruct <:CryptocomData
```
Comment (dmitrii-doronin) [inline src/Cryptocom/Spot/API/GetInstruments.jl:18]: Formatting

File: src/Cryptocom/Spot/API/Ticker.jl [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+    a::Maybe{Float64}
+    i::String
+    v::Float64
+    vv::Float64
+    oi::Float64
+    c::Maybe{Float64}
+    b::Maybe{Float64}
+    k::Maybe{Float64}
+    t::Int64
+end
```
Comment (dmitrii-doronin) [inline src/Cryptocom/Spot/API/Ticker.jl:30]: Maybe simply `Ticker` would be better?

File: src/Cryptocom/Spot/API/Candle.jl [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+end
+
+struct CandleStruct <: CryptocomData
```
Comment (dmitrii-doronin) [inline src/Cryptocom/Spot/API/Candle.jl:22]: Maybe a better name?

**Review by dmitrii-doronin — CHANGES_REQUESTED** [review]

**Review by dmitrii-doronin — APPROVED** [review]
Looks good to me!

File: src/Cryptocom/Utils.jl [hunk 15]
```diff
@@ -0,0 +1,6 @@
+#Cryptocom/Utils
+
+function Serde.SerQuery.ser_type(::Type{<:CryptocomCommonQuery}, x::D)::Int64 where {D<:DateTime}
+    return (round(Int64, datetime2unix(x))*1000)
+end
+
```
Comment (gryumov) [inline src/Cryptocom/Utils.jl:6]:
```suggestion
```

File: src/Cryptocom/Spot/API/Candle.jl [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+    x == m5  && return "5m"
+    x == m15 && return "15m"
+    x == m30 && return "30m"
+    x == h1  && return "1h"
+    x == h2  && return "2h"
+    x == h4  && return "4h"
+    x == h12 && return "12h"
+    x == d1  && return "1D"
+    x == d7  && return "7D"
+    x == d14  && return "14D"
```
Comment (gryumov) [inline src/Cryptocom/Spot/API/Candle.jl:35]:
```suggestion
    x == d14 && return "14D"
```

**Review by gryumov — APPROVED** [review]

**Pull request merged**